capture.go

```diff
@@ -0,0 +1,147 @@
+package warc
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// CaptureHelper is used for the NewRequestResponseRecords() method. Additional
+// fields may be added in the future.
+type CaptureHelper struct {
+	WarcinfoID string
+	RemoteAddr string
+
+	// The request body will need to be read multiple times, so please provide
+	// one of the following.  (note: bytes.Reader and strings.Reader are
+	// ReadSeekers.)
+	ReqBodyReadSeeker  io.ReadSeeker
+	ReqBodyBytesBuffer *bytes.Buffer
+}
+
+// DialContext returns a wrapper around net.DialContext that saves the
+// connected-to IP in the CaptureHelper.
+func (c *CaptureHelper) DialContext(dialer *net.Dialer) func(ctx context.Context, network, addr string) (net.Conn, error) {
+	if dialer == nil {
+		dialer = &net.Dialer{}
+	}
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+		conn, err := dialer.DialContext(ctx, network, addr)
+		if err != nil {
+			c.RemoteAddr = conn.RemoteAddr().String()
+		}
+		return conn, err
+	}
+}
+
+func (c *CaptureHelper) resetRequestBody(req *http.Request) io.Reader {
+	if req.Body == nil || req.Body == http.NoBody {
+		return http.NoBody
+	}
+	if req.GetBody != nil {
+		body, err := req.GetBody()
+		if err != nil {
+			panic(err)
+		}
+		return body
+	}
+	if c.ReqBodyReadSeeker != nil {
+		c.ReqBodyReadSeeker.Seek(0, io.SeekStart)
+		return c.ReqBodyReadSeeker
+	}
+	if c.ReqBodyBytesBuffer != nil {
+		return c.ReqBodyBytesBuffer
+	}
+	panic("capturehelper: unable to rewind request body")
+}
+
+// Creates a new request/response record pair for the provided HTTP request and
```

Comment:
from `golint ./...`:
`capture.go:69:1: comment on exported function NewRequestResponseRecords should be of the form "NewRequestResponseRecords ..."`